index.bs

```diff
@@ -535,66 +617,147 @@ To determine whether an |attribute| <dfn>contains a javascript: URL</dfn>:
+        list "{{SanitizerConfig/removeElements}}".
+    1. Let |result2| be [=remove an item=] |element| from |configuration| list
+        "{{SanitizerConfig/replaceWithChildrenElements}}"
+    1. Return |result1| and not |result2|.
 
 </div>
 
 <div algorithm>
 To <dfn for="Sanitizer">replace an element with its children</dfn> |element| from a {{SanitizerConfig}} |configuration|, do:
 
+1. [=Assert=]: |configuration| is [=SanitizerConfig/valid=].
 1. Set |element| to the result of [=canonicalize a sanitizer element=] with |element|.
-1. [=SanitizerConfig/Add=] |element| to |configuration|["{{SanitizerConfig/replaceWithChildrenElements}}"].
-1. [=SanitizerConfig/Remove=] |element| from |configuration|["{{SanitizerConfig/removeElements}}"].
-1. [=SanitizerConfig/Remove=] |element| from |configuration|["{{SanitizerConfig/elements}}"] list.
+1. If |configuration|["{{SanitizerConfig/replaceWithChildrenElements}}"] does
+    not [=map/exist=], then set |configuration|["{{SanitizerConfig/replaceWithChildrenElements}}"] to &laquo; [] &raquo;.
+1. [=Add an item=] |element| to |configuration|["{{SanitizerConfig/replaceWithChildrenElements}}"].
+1. If |configuration|["{{SanitizerConfig/elements}}"] [=map/exists=], then [=remove an item=] |element| from |configuration| list "{{SanitizerConfig/elements}}".
+1. If |configuration|["{{SanitizerConfig/removeElements}}"] [=map/exists=], then [=remove an item=] |element| from |configuration| list "{{SanitizerConfig/removeElements}}".
+1. If any of the [=add an item=] or [=remove an item=] calls in the last three steps returned
+    false, then return false.
+1. Return true:
 
 </div>
 
 <div algorithm>
 To <dfn for="Sanitizer">allow an attribute</dfn> |attribute| on a {{SanitizerConfig}} |configuration|, do:
 
-1. Set |attribute| to the result of [=canonicalize a sanitizer attribute=] with |attribute|.
-1. [=SanitizerConfig/Add=] |attribute| to |configuration|["{{SanitizerConfig/attributes}}"].
-1. [=SanitizerConfig/Remove=] |attribute| from |configuration|["{{SanitizerConfig/removeAttributes}}"].
+1. [=Assert=]: |configuration| is [=SanitizerConfig/valid=].
+1. Set |attribute| to the result of [=canonicalize a sanitizer attribute=] with
+    |attribute|.
+1. If |configuration|["{{SanitizerConfig/attributes}}"] [=map/exists=]:
+    1. [=Sanitizer/Add an item=] |attribute| to |configuration|
+        list "{{SanitizerConfig/attributes}}".
+    1. **IF DISALLOW REDUNDANT PER-ELEMENT-ATTRIBUTES:**
+        1. If |configuration|[{{SanitizerConfig/elements}}] [=map/exists=]:
+            1. [=list/iterate|For each=] |element| in |configuration|[{{SanitizerConfig/elements}}]:
+                1. If |element|[{{SanitizerElementNamespaceWithAttributes/attributes}}] [=map/exists=]:
+                    1. [=Sanitizer/Remove an item=] |attribute| from
+                        |element|[{{SanitizerElementNamespaceWithAttributes/attributes}}]
+    1. If the [=Sanitizer/add an item=] call returned false or
```

Comment:
I don't think we want to do this implicitly, it seems like it could be quite easy to forget to keep track of one of the return values.

Reply:
I think I agree; but haven't done the edits yet.

Reply:
We no longer us "implicit" return values.